src/ducks/Watchlists/Actions/Edit/EditForm/Assets.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useRef, useEffect, useMemo } from 'react'
+import cx from 'classnames'
+import Label from '@santiment-network/ui/Label'
+import Icon from '@santiment-network/ui/Icon'
+import Input from '@santiment-network/ui/Input'
+import Panel from '@santiment-network/ui/Panel'
+import { Checkbox } from '@santiment-network/ui/Checkboxes/Checkboxes'
+import { useTheme } from '../../../../../stores/ui/theme'
+import { useAllProjects, useOnClickOutside } from './hooks'
+import styles from './index.module.scss'
+import cardStyles from '../../../../../ducks/Watchlists/Widgets/Table/AssetCard.module.scss'
+import fieldStyles from '../../../../../ducks/Studio/Sidebar/ProjectSelector/index.module.scss'
+
+const VIEW_ITEM_COUNT = 4
+
+const Assets = ({ watchlist, onChange }) => {
+  const ref = useRef()
+  const [isSearchMode, setIsSearchMode] = useState(false)
+  const [filter, setFilter] = useState('')
+  const watchListItems = useMemo(
+    () => (watchlist ? watchlist.listItems.map(l => l.project) : []),
+    [watchlist]
+  )
+  const [items, setItems] = useState(watchListItems)
+  const watchListIDs = useMemo(() => items.map(i => i.id), [items])
+  const [checkedItems, setCheckedItems] = useState(watchListItems)
+  const { isNightMode } = useTheme()
+  const { data } = useAllProjects(filter.toLowerCase())
+  const assets = useMemo(
+    () => data.filter(item => !watchListIDs.includes(item.id)),
+    [data, watchListIDs]
+  )
+  const [showItems, setShowItems] = useState(false)
+
+  useEffect(() => {
+    const showTimeout = setTimeout(() => setShowItems(true), 300)
+    return () => clearTimeout(showTimeout)
+  }, [])
+
+  useOnClickOutside(ref, () => {
+    setFilter('')
+    setIsSearchMode(false)
+  })
+
+  useEffect(() => {
+    let _items = watchListItems
+    if (filter && filter.length > 0) {
+      const _filter = filter.toLowerCase()
+      const filterHelper = item =>
+        item.name.toLowerCase().includes(_filter) ||
+        item.ticker.toLowerCase().includes(_filter)
+      _items = _items.filter(filterHelper)
+    }
+    setItems(_items)
+  }, [filter, watchListItems])
+
+  const checkboxClickHandler = (item, newValue) =>
+    setCheckedItems(old => {
+      const items = [...old]
+      const index = items.findIndex(i => i.id === item.id)
+      if (index === -1 && newValue) {
+        items.push(item)
+      } else if (index > -1 && !newValue) {
+        items.splice(index, 1)
+      }
```

Comment:
See `[D-0-3]`

Reply:
I think here `array` are better than `set`:
if we convert items to set: `const items=new Set(old.map(i => i.id))` in the `onChange` and `setCheckedItems` we need full object not only `ids`

Reply:
You can make a set of items, not ids

Reply:
Fixed.